### DISCUSSION TITLE
Convert to int from string when writing JSON from CSV

### DISCUSSION THREAD

**Discussion by eroth** [TomWright/dasel#290] [Q&A]
Hello! I've searched far and wide for an answer to this, but can't seem to find anything. Given a CSV file with format:

```CSV
Id,Author,TotalPRsReviewed
1,billy-bob,30
2,one-sha,3,
3,tom-sawyer,69
```

When I invoke `dasel -f ~/Desktop/PR\ reviewer\ size.csv -w json 'all().mapOf(id,Id,name,Author,numPRs,TotalPRsReviewed).merge()'` the output is:

```JSON
[
  {
    "id": "1",
    "name": "billy-bob,",
    "numPRs": "30"
  },
  {
    "id": "2",
    "name": "one-sha",
    "numPRs": "3"
  },
  {
    "id": "3",
    "name": "tom-sawyer",
    "numPRs": "69"
  }
]
```

As can be seen, each JSON field value is a string. I'm guessing this is the underlying or default type for CSV files. What I'd like is if the `id` field could be an `int` (I don't care about `numPRs`, it's for display). I've seen an [example](https://daseldocs.tomwright.me/commands/put) using the `Put` command along with the `--t` flag but don't see how that can be used alongside what I'm doing. Any suggestions? Thanks!

**Comment by TomWright** [comment]
Frustratingly I can't think of a solution to this either.

I think a simple solution may be a `toType` function that accept a type (string, int, float, bool etc) and a value. It will try to convert between the values as best it can.

**Reply by eroth** [reply]
Is that difficult to do? Is it something I could implement? I haven't really looked into dasel's codebase yet.

**Reply by TomWright** [reply]
It shouldn't be too difficult.

If you are interested in making a PR I'll happily accept one, but I'm also happy to implement this feature when I find a natural stopping point in the other changes I'm making.

**Reply by eroth** [reply]
I haven't done any Go development before and wouldn't know where to begin as far as setting up an environment goes. I also wouldn't know where to begin as far as making this change, so it should probably wait until you have time— unless you want me asking you a bunch of questions 😆
